sphinx/tutorial/rose/furthertopics/rose-arch.rst

```diff
@@ -0,0 +1,281 @@
+----------------------
+
+Change to the ``$ROSE_DATA/ARCHVING/`` directory of the suite i.e::
+
+   cd ~cylc-run/SUITE_ID/share/data/ARCHIVING/
+
+List the directory by typing::
+
+   ls
+
+You should see the following returned:
+
+.. code-block:: none
+
+   dir  file_zipped.tar  files  rename  solo.file
+
+Change directory to ``files/`` and list the files::
+
+   cd files
+   ls
+
+The following should be returned:
+
+.. code-block:: none
+
+   file_list1  file_list3
+
+Change directory to ``ARCHIVING/dir/`` and list the files::
+
+   cd ..
+   cd dir
+   ls
+
+The following should be returned:
+
+.. code-block:: none
+
+   file_list1  file_list2 file_list3
+
+.. note::
+
+   These were all of the files in the ``$ROSE_DATA/`` directory.
+
+Change diectory to ``ARCHIVING/rename/`` and list the files::
+
+   cd ..
+   cd rename
+   ls
+
+The following should be returned:
+
+.. code-block:: none
+
+   1_1_file_list1 1_2_file_list2 1_3_file_list3 
+
+These are the renamed files.
+
+Most users will have there own system or location that they wish to archive
+their data to. Here the example shown uses ``rsync`` and ``scp``. Further
+information on Linux commands ``rsync``, ``scp`` Please refer your own site
+specific archiving solutions and seek site specfic advice.
+
+
+Arch Settings
+-------------
+
+Some settings that can be used are described below.
+
+.. TODO - link to reference material
+
+   Also see: Built-in Applications: rose_arch
+
+Above ``.tar`` was used to compress the file. However, ``compress=gzip``
+can also be used. Note either of these commands can be used to compress a
+file or a folder/directory.
+
+In the above example a regular expression (reg exp) was used by the
+``rename-parser``, for example, ``^.*list(?P<tag>.*)$``, where:
+
+* ``^`` = startof a string.
+* ``$`` = end of a string.
+* ``.`` = any character.
+* ``*`` = greedy (all).
+* ``?P<NAME>`` = named group.
+
+.. note::
+
+   .. _python flavor: https://docs.python.org/2/howto/regex.html
+
+   ``rose arch`` uses the `python flavor`_ for regular expressions.
+
+In the above example source was used to accept a list of glob patterns.
+For example, ``file_list?`` was used where the ``?`` relates to one unknown
+character. Similarly, if there were files named ``pa000``, ``pa012``,
+``pa018``, ``pa024`` which are to be source files can be identified using
+``source=pa???`` where the ``???`` relate to the three possible number
+characters.
+
+.. note::
+
+   These examples are just some possible examples and not a full list.
+
+As well as ``[arch]`` and ``[arch:TARGET]`` other options can be provided
+to the app, for example:
+
+``[env]``
+   Can be defined near the top of the app to allow an environment variable
+   to be available to the ``[arch:]`` commands in the app.
+
+   Also see the suite example above.
+``[poll]``
+   Polling can be defined, and is often near the bottom of the app. This
+   will allow the app to poll with a defined delay, e.g. ``delay=0``.
```

Comment:
Not relevant to the tutorial, but please change the ``delay`` variable example from ``0`` as a 'delay' of zero is a bit of a perplexing distraction when reading through this.